src/hypothesis/extra/pandas/impl.py

```diff
@@ -70,6 +104,136 @@ def dtype_for_elements_strategy(s):
     )
 
 
+class IndexStrategy(st.SearchStrategy):
+    def __init__(
+        self, elements, dtype, min_size, max_size, unique, order,
+        allow_nan
+    ):
+        super(IndexStrategy, self).__init__()
+        self.elements = elements
+        self.dtype = dtype
+        self.min_size = min_size
+        self.max_size = max_size
+        self.unique = unique
+        self.order = order
+        self.allow_nan = allow_nan
+
+    def do_draw(self, data):
+        result = []
+        seen = set()
+
+        iterator = cu.many(
+            data, min_size=self.min_size, max_size=self.max_size,
+            average_size=(self.min_size + self.max_size) / 2
+        )
+
+        if self.dtype is None:
+            def convert(x): return x
+        else:
+            convert = self.dtype.type
+
+        while iterator.more():
+            elt = convert(data.draw(self.elements(len(result))))
+
+            if not self.allow_nan:
+                try:
+                    if np.isnan(float(elt)):
+                        iterator.reject()
+                        continue
+                except TypeError:
+                    pass
+
+            if self.order != 0 and result:
+                if (
+                    (elt > result[-1] and self.order < 0) or
+                    (elt < result[-1] and self.order > 0)
+                ):
+                    iterator.reject()
+                    continue
+
+            if self.unique:
+                if elt in seen:
+                    iterator.reject()
+                    continue
+                seen.add(elt)
+            result.append(elt)
+
+        if not result and self.dtype is None:
+            dtype = data.draw(dtype_for_elements_strategy(self.elements(0)))
+            return pandas.Index([], dtype=dtype)
+        return pandas.Index(result, dtype=self.dtype)
+
+
+@st.cacheable
+@st.defines_strategy
+def indexes(
```

Comment:
Does wanting this to be called `indices()` make me a bad person?  I just twitch every time I see "indexes"...